docs/shared/plugins/intro.md

```diff
@@ -1,58 +1,108 @@
+{% cards cols="2" %}
+
+{% link-card title="Enforce Best Practices in Your Repository" type="tutorial" url="/extending-nx/tutorials/organization-specific-plugin" icon="office" /%}
+{% link-card title="Integrate a Tool Into an Nx Repository" type="tutorial" url="/extending-nx/tutorials/tooling-plugin" icon="tool" /%}
 
 {% /cards %}
 
-## Maintain a Published Plugin
+## Create Your First Code Generator
 
-If your plugin has functionality that would be useful in more than just your repo, you can publish it to npm and register it on the nx.dev site for others to find.
+Wire up a new generator with this terminal command:
 
-{% cards cols="2" %}
+```shell
+npx nx g generator library-with-readme --directory=packages/my-plugin/src/generators/library-with-readme
+```
 
-{% card title="Share Your Plugin" description="Submit your plugin to the Nx plugin registry" url="/extending-nx/tutorials/publish-plugin" /%}
-{% card title="Migration Generators" description="Update repos when you introduce breaking changes" url="/extending-nx/recipes/migration-generators" /%}
+### Understand the Generator Functionality
 
-{% /cards %}
+This command will register the generator in the plugin's `generators.json` file and create some default generator code in the `library-with-readme` folder. The `libraryWithReadmeGenerator` function in the `generator.ts` file is where the generator functionality is defined.
 
-## Advanced Plugins
+```typescript {% fileName="packages/my-plugin/src/generators/library-with-readme/generator.ts" %}
+export async function libraryWithReadmeGenerator(
+  tree: Tree,
+  options: LibraryWithReadmeGeneratorSchema
+) {
+  const projectRoot = `libs/${options.name}`;
+  addProjectConfiguration(tree, options.name, {
+    root: projectRoot,
+    projectType: 'library',
+    sourceRoot: `${projectRoot}/src`,
+    targets: {},
+  });
+  generateFiles(tree, path.join(__dirname, 'files'), projectRoot, options);
+  await formatFiles(tree);
+}
+```
 
-You can also hook into the way Nx works and modify it to suit your needs
+This generator calls the following functions:
 
-{% cards cols="2" %}
+- `addProjectConfiguration` - Create a new project configured for TypeScript code.
+- `generateFiles` - Create files in the new project based on the template files in the `files` folder.
+- `formatFiles` - Format the newly created files with Prettier.
 
-{% card title="Scaffold a New Workspace" description="Set up a new repo" url="/extending-nx/recipes/create-preset" /%}
-{% card title="Project Graph Plugins" description="Modify the Nx graph" url="/extending-nx/recipes/project-graph-plugins" /%}
+You can find more helper functions in the [Nx Devkit reference documentation](/nx-api/devkit/documents/nx_devkit).
 
-{% /cards %}
+### Create a README Template File
+
+We can remove the unnecessary `index.ts.template` file and add our own `README.md.template` file in the `files` folder.
```

Comment:
I'd remove the word "unnecessary" so it doesn't sound as negative. Maybe "We can remove the generated `index.ts.template` file and add our own..."